streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryTimeOrderedKeyValueBuffer.java

```diff
@@ -181,10 +186,25 @@ public void setSerdesIfNull(final Serde<K> keySerde, final Serde<V> valueSerde)
 
     @Override
     public void init(final ProcessorContext context, final StateStore root) {
+        taskId = context.taskId().toString();
         final InternalProcessorContext internalProcessorContext = (InternalProcessorContext) context;
-
-        bufferSizeSensor = Sensors.createBufferSizeSensor(this, internalProcessorContext);
-        bufferCountSensor = Sensors.createBufferCountSensor(this, internalProcessorContext);
+        streamsMetrics = internalProcessorContext.metrics();
+
+        threadId = Thread.currentThread().getName();
+        bufferSizeSensor = StateStoreMetrics.suppressionBufferSizeSensor(
+            threadId,
+            taskId,
+            METRIC_SCOPE,
+            storeName,
+            streamsMetrics
+        );
+        bufferCountSensor = StateStoreMetrics.suppressionBufferCountSensor(
+            threadId,
+            taskId,
+            METRIC_SCOPE,
+            storeName,
+            streamsMetrics
+        );
```

Comment:
Metrics for suppression buffers are fetched in this way now.

Reply:
Sounds good, I think not relying on context but passing in parameters explicitly is fine (if that's what you mean has changed here).

Reply:
I wanted just to highlight the most important parts of the PR.

Actually, I would prefer to add the thread ID to the context and use the context for the metrics. I have already tried to do it, but I ran into a `ClassCastException` because we cast `ProcessorContext` to `InternalProcessorContext` (which IMO should be changed in future if possible). Using the context instead of explicit parameters makes the signature of the methods more robust. These rather minor issue atm.